### ISSUE TITLE
Fatal Exception: java.lang.IllegalStateException: Instabug API called before Instabug.Builder().build() was called

### ISSUE THREAD

**Issue by gozozo** [Instabug/Instabug-Android#120]
### Steps to Reproduce the Problem
Remove app and install again, the error doesn't permit open the app

### Instabug integration code
 `Instabug.setLocale(new Locale("es"));`

 `new Instabug.Builder(this.getApplication(), "token")
     .setInvocationEvents(InstabugInvocationEvent.SHAKE,InstabugInvocationEvent.SCREENSHOT)
     .build();`

### Error

`Fatal Exception: java.lang.IllegalStateException: Instabug API called before Instabug.Builder().build() was called
       at com.instabug.library.APIBuildChecker.check(APIBuildChecker.java:6)
       at com.instabug.library.Instabug.getAppToken(Instabug.java:305)
       at com.instabug.library.network.NetworkManager.buildRequest(NetworkManager.java:180)
       at com.instabug.library.network.NetworkManager.buildRequest(NetworkManager.java:154)
       at com.instabug.library.network.NetworkManager.buildRequest(NetworkManager.java:145)
       at com.instabug.library.network.a.a.a(FeaturesService.java:41)
       at com.instabug.library.b.c(InstabugFeaturesManager.java:238)
       at com.instabug.library.b$2.run(InstabugFeaturesManager.java:185)
       at java.lang.Thread.run(Thread.java:818)`

**Comment by eygraber** [comment]
I've had the same issue since updating to Instabug 8.0.5.

This is what my instabug code looks like. It is called when the app starts.

```
Instabug.Builder(app, BuildConfig.INSTABUG_TOKEN)
        .setInAppMessagingState(Feature.State.ENABLED)
        .setSurveysState(Feature.State.ENABLED)
        .setCrashReportingState(Feature.State.DISABLED)
        .setInstabugLogState(Feature.State.DISABLED)
        .setConsoleLogState(Feature.State.DISABLED)
        .setPushNotificationState(Feature.State.DISABLED)
        .setTrackingUserStepsState(Feature.State.DISABLED)
        .setUserEventsState(Feature.State.DISABLED)
        .setReproStepsState(State.DISABLED)
        .build(Feature.State.ENABLED)

    Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED)
    Instabug.setShouldPlayConversationSounds(false)

    BugReporting.setInvocationEvents(InstabugInvocationEvent.SHAKE)

    // enabled is persisted in SharedPreferences and defaults to true
    if(!enabled) {
      Instabug.disable()
    }
```

Crash is:

```
Fatal Exception: java.lang.IllegalStateException: Instabug API called before Instabug.Builder().build() was called
       at com.instabug.library.APIBuildChecker.check(SourceFile:6)
       at com.instabug.library.Instabug.getAppToken(SourceFile:305)
       at com.instabug.library.network.NetworkManager.buildRequest(SourceFile:180)
       at com.instabug.library.network.NetworkManager.buildRequest(SourceFile:154)
       at com.instabug.library.network.NetworkManager.buildRequest(SourceFile:145)
       at com.instabug.library.network.a.a.a(SourceFile:41)
       at com.instabug.library.b.c(SourceFile:238)
       at com.instabug.library.b$2.run(SourceFile:185)
       at java.lang.Thread.run(Thread.java:762)
```

I've seen the crash occur on Android 6 and Android 7 devices, on Motorola, LG, and Samsung.

**Comment by rdiaa** [comment]
@gozozo @eygraber 

The issue is fixed and we released a new Snapshot with the fix. please install 8.0.5.1-SNAPSHOT , the crash is fixed here.

**Comment by Kevinrob** [comment]
@rdiaa Does 8.0.6 contains this fix?

**Comment by rdiaa** [comment]
@Kevinrob 

Yes,  the fix is included in the new version "8.0.6"
